commons/src/main/scala/za/co/absa/spline/common/StringEscapeUtils.scala

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2022 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spline.common
+
+object StringEscapeUtils {
+  def escapeAQLSearch(str: String): String = str
+    .replaceAll("_", "\\\\\\\\_")
+    .replaceAll("%", "\\\\\\\\%")
```

Comment:
That is a lot of backslashes. Maybe [Matcher.quoteReplacement](https://docs.oracle.com/javase/7/docs/api/java/util/regex/Matcher.html#quoteReplacement(java.lang.String)) could be used here as suggested in replaceAll docs. To make it more obvious what is happening.

Reply:
Not sure if this makes it more obvious :)
```scala
  def escapeAQLSearch(str: String): String = {
    str
      .replaceAll("_", quoteReplacement(quoteReplacement("\\_")))
      .replaceAll("%", quoteReplacement(quoteReplacement("\\%")))
  }
```
What about this then?
```scala
  def escapeAQLSearch(str: String): String = {
    str
      .replaceAll("_", """\\\\_""")
      .replaceAll("%", """\\\\%""")
  }

```

Reply:
What about this:

```scala
  def escapeAQLSearch(str: String): String = {
    str
      .replace("_", """\\_""")
      .replace("%", """\\%""")
  }
```

It seems to be dooing exactly what you want without the need for regexp replacement escaping.

Reply:
Lol I must be too enthusiastic about regexps :rofl: